dimod/reference/composites/fixedvariable.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2018 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+"""
+A composite that fixes the variables provided and removes them from the
+bqm object before sending to its child sampler.
+"""
+import numpy as np
+from dimod.response import SampleSet
+
+import dimod
+
+__all__ = ['FixedVariableComposite']
+
+
+class FixedVariableComposite(dimod.ComposedSampler):
+    """Composite to fix variables of a problem to provided, assigned values
+
+    Inherits from :class:`dimod.ComposedSampler`.
+
+    Fixes variables of a bqm and modifies linear and quadratic terms
+    accordingly. Returned samples include the fixed variable
+
+    Args:
+       sampler (:class:`dimod.Sampler`):
+            A dimod sampler
+
+    Examples:
+       This example uses :class:`.FixedVariableComposite` to instantiate a
+       composed sampler that submits a simple Ising problem to a sampler.
+       The composed sampler fixes a variable and modifies linear and quadratic
+       biases according.
+
+       >>> from dwave.system.samplers import DWaveSampler
```

Comment:
This example uses a library potentially not available to the user. I think we should just show it wrapping the exact solver but maybe we should talk about that.